src/test/scala/mesosphere/marathon/core/instance/TestInstanceBuilder.scala

```diff
@@ -31,12 +31,6 @@ case class TestInstanceBuilder(instance: Instance, now: Timestamp = Timestamp.no
   def addTaskLost(since: Timestamp = now, containerName: Option[String] = None): TestInstanceBuilder =
     addTaskWithBuilder().taskLost(since, containerName).build()
 
-  def addTaskReserved(containerName: Option[String] = None): TestInstanceBuilder =
-    addTaskWithBuilder().taskReserved(containerName).build()
```

Comment:
As I mentioned above we still have to have these. We did not remove reserved tasks.

Reply:
these builders just created instances and tasks with condition reserved, this is really obsolete and we don't need that anymore since we dropped that condition.

Reply:
Hm, how is a `TestTaskBuilder.taskResidentLaunched` different from `TestTaskBuilder.taskLaunched`? Seems we should `TestTaskBuilder.taskResidentLaunched` as well then.

Reply:
`taskResidentLaunched` creates *instance with reservation* and a task that is running. I actually think that we are mixing two things there because reservation is really not a property of task. But I don't think it makes sense to invest more into those builders other than removing them and replacing them with production code

Reply:
Hm, why not fix it? It's not a blocker for this PR but should it be 

```scala
TestInstanceBuilder(...).withReservation(...).addRunningTask()
```
then?